src/battle_util.c

```diff
@@ -6313,7 +6313,7 @@ bool32 IsBattlerTerrainAffected(u8 battler, u32 terrainFlag)
     return IsBattlerGrounded(battler);
 }
 
-bool32 CanSleep(u8 battler)
+bool32 CanSleep(u32 battler)
```

Comment:
would updating these CanBeStatused functions to take ability, hold effect etc speed up AI calcs?

Reply:
I think so, tho not sure if it would be enough to get a single frame.